dkg-primitives/src/types.rs

```diff
@@ -259,6 +259,7 @@ pub enum DKGError {
 	NotListeningForPublicKey,
 	NoAuthorityAccounts,
 	NoHeader,
+	MPCSignError { reason: String, bad_actors: Vec<usize> },
```

Comment:
If you're adding MPCSignError and using it instead of `SignMisbehaviour`, can you do the same with the Keygen error too and remove their counterparts too?